### PR TITLE
New Apm config options

### PR DESCRIPTION
**Pull request by turtledreams**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## X.X.X
+*
```
Comment (arifBurakDemiray) [inline CHANGELOG.md:2]: Is entry missing?

File: android/src/main/java/ly/count/dart/countly_flutter/CountlyFlutterPlugin.java [hunk 2]
```diff
@@ -135,6 +135,13 @@ public final int resultResponder(RequestResult rResult) {
 
     List<CountlyFeedbackWidget> retrievedWidgetList = null;
 
+    //-------------TESTING RELATED------------------
+    private static Boolean isAppStartTimeTracked = false;
+    private static Boolean isFBEnabled = false;
```
Comment (arifBurakDemiray) [inline android/src/main/java/ly/count/dart/countly_flutter/CountlyFlutterPlugin.java:140]: isForegroundBackgroundEnabled is shorther than the other 
isManualAppLoadedTriggerEnabled

I think this reduces readibility

File: lib/src/configuration_interfaces/countly_config_apm.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  /// Enables the automatic tracking of app foreground and background durations.
+  CountlyConfigApm enableForegroundBackgroundTracking() {
+    _enableForegroundBackground = true;
+    return this;
+  }
+
+  /// Enables the automatic tracking of app start time.
+  CountlyConfigApm enableAppStartTimeTracking() {
+    _recordAppStartTime = true;
```
Comment (arifBurakDemiray) [inline lib/src/configuration_interfaces/countly_config_apm.dart:26]: to comply with the function name, maybe _trackAppStartTime but just an idea

**Review by arifBurakDemiray — CHANGES_REQUESTED** [review]
Couple of things maybe updated

File: CHANGELOG.md (hunk 1 above)
Comment (turtledreams) [reply CHANGELOG.md:2]: yeap still cooking

File: lib/src/configuration_interfaces/countly_config_apm.dart (hunk 3 above)
Comment (turtledreams) [reply lib/src/configuration_interfaces/countly_config_apm.dart:26]: i think so too. keeping the old one but changing looks more appealing indeed

**Review by arifBurakDemiray — APPROVED** [review]

**Review by arifBurakDemiray — APPROVED** [review]

File: lib/src/configuration_interfaces/countly_config_apm.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  /// Enables manual trigger of the moment when the app has finished loading.
+  CountlyConfigApm enableManualAppLoadedTrigger() {
+    _enableManualAppLoaded = true;
+    return this;
+  }
+
+  /// Gives you the ability to override the automatic app start timestamp.
+  CountlyConfigApm setAppStartTimestampOverride(int timestamp) {
```
Comment (ArtursKadikis) [inline lib/src/configuration_interfaces/countly_config_apm.dart:37]: not enough explanation of the param

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

**Review by arifBurakDemiray — APPROVED** [review]

**Pull request merged**